<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="deadzone_val"   default="0.1" />

    <arg name="button_map_name"   default="button_mapping_1.yaml" />
    <arg name="axis_map_name"   default="axis_mapping_2.yaml" />

    <node name="$(anon joystick)" pkg="joy" type="joy_node" output="screen" required="false">
        <param name="dev" value="$(arg device)" type="str"/>
        <param name="deadzone" value="0.0" />
        <param name="autorepeat_rate" value="30.0" />
        <remap from="joy" to="/$(arg topic_name)" />
        <remap from="joy/set_feedback" to="/$(arg topic_name)/set_feedback" />
    </node>
    <node name="$(anon joy_throttle)" pkg="topic_tools" type="throttle" required="false" args="messages /$(arg topic_name) 20 /$(arg topic_name)_throttled" />


    <group ns="dodobot" >
        <rosparam file="/home/ben/.local/dodobot/dodobot_py/config/joystick.yaml" command="load" ns="joystick"/>

        <node name="db_joystick_parsing" pkg="db_joystick" type="parsing_joystick.py" output="screen" required="false">
            <param name="deadzone_joy_val" value="$(arg deadzone_val)" />
            <param name="joystick_topic" value="/$(arg topic_name)_throttled" />
            <param name="linear_scale" value="0.69" />
            <param name="angular_scale" value="6.0" />
            <param name="gripper_max_dist" value="0.112" />

            <rosparam param="button_mapping" file="$(find tj2_debug_joystick)/config/$(arg button_map_name)" command="load"/>
            <rosparam param="axis_mapping" file="$(find tj2_debug_joystick)/config/$(arg axis_map_name)" command="load"/>

            <param name="linear_axis" value="left/Y" />
            <param name="angular_axis" value="left/X" />
            <param name="stepper_axis" value="right/Y" />
            <param name="tilter_axis" value="right/Y" />

            <param name="joystick_mode" value="1" />
            <remap from="cmd_vel" to="cmd_vel_teleop"/>
        </node>
    </group>
</launch>
